phoenix-tracing-webapp/src/main/java/org/apache/phoenix/tracingwebapp/http/TraceServlet.java

```diff
@@ -0,0 +1,160 @@
+    String action = request.getParameter("action");
+    String limit = request.getParameter("limit");
+    String traceid = request.getParameter("traceid");
+    String parentid = request.getParameter("parentid");
+    String jsonObject = "{}";
+    if ("getall".equals(action)) {
+      jsonObject = getAll(limit);
+    } else if ("getCount".equals(action)) {
+      jsonObject = getCount("description");
+    } else if ("getDistribution".equals(action)) {
+      jsonObject = getCount(DEFAULT_COUNTBY);
+    } else if ("searchTrace".equals(action)) {
+      jsonObject = searchTrace(parentid, traceid, LOGIC_OR);
+    } else {
+      jsonObject = "{ \"Server\": \"Phoenix Tracing Web App\", \"API version\": 0.1 }";
+    }
+    //response send as json
+    response.setContentType("application/json");    
+    String output = jsonObject;
+    PrintWriter out = response.getWriter();
+    out.print(output);
+    out.flush();
+
+  }
+
+  //get all trace results with limit count
+  protected String getAll(String limit) {
+    String json = null;
+    if(limit == null) {
+      limit = DEFAULT_LIMIT;
+    }
+    String sqlQuery = "SELECT * FROM " + TRACING_TABLE + " LIMIT "+limit;
+    json = getResults(sqlQuery);    
+    return getJson(json);
+  }
+
+  //get count on traces can pick on param to count
+  protected String getCount(String countby) {
+    String json = null;
+    if(countby == null) {
+      countby = DEFAULT_COUNTBY;
+    }
+    String sqlQuery = "SELECT "+countby+", COUNT(*) AS count FROM " + TRACING_TABLE + " GROUP BY "+countby+" HAVING COUNT(*) > 1 ";
+    json = getResults(sqlQuery);
+    return json;
+  }
+
+  //search the trace over parent id or trace id
+  protected String searchTrace(String parentId, String traceId,String logic) {
+    String json = null;
+    String query = null;
+    if(parentId != null && traceId != null) {
+      query = "SELECT * FROM " + TRACING_TABLE + " WHERE parent_id="+parentId+" "+logic+" trace_id="+traceId;
+    }else if (parentId != null && traceId == null) {
+      query = "SELECT * FROM " + TRACING_TABLE + " WHERE parent_id="+parentId;
+    }else if(parentId == null && traceId != null) {
+      query = "SELECT * FROM " + TRACING_TABLE + " WHERE trace_id="+traceId;
+    }
+    json = getResults(query);
+    return getJson(json);
+  }
+
+  //return json string
+  protected String getJson(String json) {
+    String output = json.toString().replace("_id\":", "_id\":\"")
+        .replace(",\"hostname", "\",\"hostname")
+        .replace(",\"parent", "\",\"parent")
+        .replace(",\"end", "\",\"end");   
+    return output;
+  }
+
+  //get results with passing sql query
+  protected String getResults(String sqlQuery) {
+    String json = null;
+    if(sqlQuery == null){
+    json = "{error:true,msg:'SQL was null'}";
+    }else{
+    try {
+      Class.forName("org.apache.phoenix.jdbc.PhoenixDriver");
+      con = DriverManager.getConnection("jdbc:phoenix:"+PHOENIX_HOST+":"+PHOENIX_PORT);
```

Comment:
connection should be reused, not re-created with every query.

phoenix host and port should be read from a configuration object. It will need to be provided from the parsing done in Main.